src/decoding/extrinsic_decoder.rs

```diff
@@ -559,18 +565,31 @@ where
         0b00 => ExtrinsicType::Bare,
         0b10 => ExtrinsicType::Signed,
         0b01 => ExtrinsicType::General,
-        _ => return Err(ExtrinsicDecodeError::VersionTypeNotSupported(version_type)),
+        _ => {
+            return Err(ExtrinsicDecodeError::ExtrinsicTypeNotSupported {
+                version,
+                extrinsic_type: version_type,
+            })
+        }
     };
 
-    let curr_idx = |cursor: &mut &[u8]| (bytes.len() - cursor.len()) as u32;
+    // V5 extrinsics have no notion of "signed" extrinsics.
+    if version == 5 && version_ty == ExtrinsicType::Signed {
```

Comment:
nit: move this to line 566 as a conditional branch in pattern?

Reply:
That's a nice little improvement!